src/main/java/bio/terra/cli/command/resources/Delete.java

```diff
@@ -1,27 +1,72 @@
 public class Delete extends BaseCommand {
-
   @CommandLine.Option(
       names = "--name",
       required = true,
-      description = "The name of the resource, scoped to the workspace.")
+      description = "Name of the resource, scoped to the workspace.")
   private String name;
 
-  /** Delete an existing controlled resource. */
+  @CommandLine.Mixin FormatOption formatOption;
+
+  /** Delete a resource from the workspace. */
   @Override
   protected void execute() {
-    CloudResource resource =
-        new WorkspaceManager(globalContext, workspaceContext).deleteControlledResource(name);
-    OUT.println(resource.type + " successfully deleted: " + resource.cloudId);
-    OUT.println("Workspace resource successfully removed: " + resource.name);
+    // get the resource summary object
+    WorkspaceManager workspaceManager = new WorkspaceManager(globalContext, workspaceContext);
+    ResourceDescription resourceToDelete = workspaceManager.getResource(name);
+
+    // call the appropriate delete endpoint for the resource
+    // there is a different endpoint(s) for deleting each combination of resource type and
+    // stewardship type, but all require only the workspace and resource unique ids, so calling them
+    // looks very similar from the CLI user's perspective
+    if (resourceToDelete.getMetadata().getStewardshipType().equals(StewardshipType.REFERENCED)) {
+      switch (resourceToDelete.getMetadata().getResourceType()) {
+        case GCS_BUCKET:
+          workspaceManager.deleteReferencedGcsBucket(name);
+          break;
+        case BIG_QUERY_DATASET:
+          workspaceManager.deleteReferencedBigQueryDataset(name);
+          break;
+        default:
+          throw new UnsupportedOperationException(
+              "Resource type not supported: " + resourceToDelete.getMetadata().getResourceType());
+      }
+    } else if (resourceToDelete
+        .getMetadata()
+        .getStewardshipType()
+        .equals(StewardshipType.CONTROLLED)) {
+      switch (resourceToDelete.getMetadata().getResourceType()) {
+        case GCS_BUCKET:
+          workspaceManager.deleteControlledGcsBucket(name);
+          break;
+        case BIG_QUERY_DATASET:
+          workspaceManager.deleteControlledBigQueryDataset(name);
+          break;
+        default:
+          throw new UnsupportedOperationException(
+              "Resource type not supported: " + resourceToDelete.getMetadata().getResourceType());
+      }
+    } else {
+      throw new UnsupportedOperationException(
+          "Stewardship type not supported: " + resourceToDelete.getMetadata().getStewardshipType());
+    }
+
+    formatOption.printReturnValue(resourceToDelete, Delete::printText);
+  }
+
+  /** Print this command's output in text format. */
+  private static void printText(ResourceDescription returnValue) {
+    OUT.println("Successfully deleted resource.");
+    bio.terra.cli.command.resources.Describe.printText(returnValue);
```

Comment:
I think it might be nice to have a central file for defining the ways to print the different models returned by WM. I would guess we eventually want to print the same models for multiple commands. While it's fine for commands to have their own custom printing of models as needed, a central place would help promote code re-use.

Reply:
Agreed about a central place being helpful.
Created a new `PrintingUtils` class with two methods so far:
- `printResource` which is just this method relocated
- `printWorkspace` which I added to several workspace-related commands `status`, `workspace create`, `workspace update`, `workspace delete`, `workspace mount`.